controllers/scheduler_test.go

```diff
@@ -2,291 +2,518 @@ package controllers_test
-				Expect(err).ToNot(HaveOccurred())
+				It("sets the scheduling conditions on the Work", func() {
+					Expect(fakeK8sClient.Get(context.Background(), client.ObjectKeyFromObject(&resourceWork), &resourceWork))
+					Expect(resourceWork.Status.Conditions).To(HaveLen(2))
 
-				Expect(k8sClient.List(context.Background(), &workPlacements)).To(Succeed())
+					Expect(resourceWork.Status.Conditions[0].Type).To(Equal("Scheduled"))
+					Expect(resourceWork.Status.Conditions[0].Status).To(Equal(v1.ConditionTrue))
+					Expect(resourceWork.Status.Conditions[0].Message).To(Equal("All WorkloadGroups scheduled to Destination(s)"))
+					Expect(resourceWork.Status.Conditions[0].Reason).To(Equal("ScheduledToDestinations"))
 
-				Expect(workPlacements.Items).To(HaveLen(1))
-				workPlacement = workPlacements.Items[0]
-				Expect(workPlacement.Spec.Workloads).To(HaveLen(2))
-				Expect(workPlacement.Spec.Workloads).To(ContainElement(Workload{
-					Content: "fake: content",
-				}))
+					Expect(resourceWork.Status.Conditions[1].Type).To(Equal("Misscheduled"))
+					Expect(resourceWork.Status.Conditions[1].Status).To(Equal(v1.ConditionFalse))
+					Expect(resourceWork.Status.Conditions[1].Message).To(Equal("WorkGroups that have been scheduled are at the correct Destination(s)"))
+					Expect(resourceWork.Status.Conditions[1].Reason).To(Equal("ScheduledToCorrectDestinations"))
+				})
 			})
-			When("An update removes a workload group", func() {
-				It("removes the workplacements for the deleted workloadgroup", func() {
+
+			When("a resource Work with scheduling is reconciled twice", func() {
+				var workPlacement WorkPlacement
+
+				BeforeEach(func() {
 					_, err := scheduler.ReconcileWork(&resourceWork)
-					Expect(k8sClient.Get(context.Background(), client.ObjectKeyFromObject(&resourceWork), &resourceWork))
 					Expect(err).ToNot(HaveOccurred())
 
-					Expect(k8sClient.List(context.Background(), &workPlacements)).To(Succeed())
+					latestWork := &Work{}
+					Expect(fakeK8sClient.Get(context.Background(), client.ObjectKeyFromObject(&resourceWork), latestWork)).To(Succeed())
+
+					Expect(fakeK8sClient.List(context.Background(), &workPlacements)).To(Succeed())
+					Expect(workPlacements.Items).To(HaveLen(1))
+					workPlacement = workPlacements.Items[0]
+
+					_, err = scheduler.ReconcileWork(latestWork)
+					Expect(err).ToNot(HaveOccurred())
+				})
 
+				It("does not change the existing WorkPlacement", func() {
+					Expect(fakeK8sClient.List(context.Background(), &workPlacements)).To(Succeed())
 					Expect(workPlacements.Items).To(HaveLen(1))
+					newWorkPlacement := workPlacements.Items[0]
+
+					// compare each field in the work placements to ensure they are the same
+					Expect(newWorkPlacement.Name).To(Equal(workPlacement.Name))
+					Expect(newWorkPlacement.Namespace).To(Equal(workPlacement.Namespace))
+					Expect(newWorkPlacement.ObjectMeta.Labels["kratix.io/work"]).To(Equal(workPlacement.ObjectMeta.Labels["kratix.io/work"]))
```

Comment:
nit: very long lines... should we break?

Reply:
alternative:

```
Expect(newWorkPlacement.Labels["kratix.io/work"]).To(Equal(workPlacement.Labels["kratix.io/work"]))
```

Reply:
if all labels match:
```go
Expect(newWorkPlacements.Labels).To(Equal(workPlacement.Labels["kratix.io/work"]))
```

Reply:
Created [186726492](https://www.pivotaltracker.com/story/show/186726492) to start using gstruct ✅